api/repositories.py

```diff
@@ -248,18 +222,19 @@ def get_all_devices(self):
 
     def set_power_state(self, device_id, power_state):
         device = self.get_device_by_id(device_id)
-        assert(device.device_type == "light_switch"), "Device is not a switch."
+        if device.device_type != "light_switch":
+            raise Exception("Device is not a switch.")
         device.configure_power_state(power_state)
         self.update_device_reading(device)
         self.collection.update_one({'_id': device_id}, {"$set": {'power_state': power_state}}, upsert=False)
 
     def set_target_temperature(self, device_id, temp):
         device = self.collection.find_one({'_id': device_id})
         assert (device['device_type'] == "thermostat"), "Device is not a thermostat."
-        assert ('locked_min_temperature' not in device or device[
-            'locked_min_temperature'] <= temp), "Chosen temperature is too low."
-        assert ('locked_max_temperature' not in device or device[
-            'locked_max_temperature'] >= temp), "Chosen temperature is too high."
+        if device['locked_min_temperature'] > temp:
```

Comment:
That's not a reason for our application to crash. It is definitely a fault in the device data, but it shouldn't raise an Exception.